sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -596,11 +597,13 @@ class RelationalGroupedDataset protected[sql](
       s"The returnType of the udf must be a ${StructType.simpleString}")
 
     val leftGroupingNamedExpressions = groupingExprs.map {
+      case a: Attribute => Alias(a, a.name)()
       case ne: NamedExpression => ne
       case other => Alias(other, other.toString)()
     }
 
     val rightGroupingNamedExpressions = r.groupingExprs.map {
+      case a: Attribute => Alias(a, a.name)()
```

Comment:
These changes are needed to make the output of `Project`s in line 615/617 unique. Without these the new assert in `removeRedundantAliases()` would fail.

Reply:
hmmm, does it mean it's legitimate to have conflicting attribute IDs within one operator? This looks totally fine in `FlatMapCoGroupsInPandas`

Reply:
I don't know, this was the only place where I had to modify the code to avoid conflicts.
But I was thinking about that instead of the new `assert` in the rule, we could fix these conflicting attributes with inserting a `Project`. How about that?